### PR TITLE
MINOR: Various cleanups in coordinator modules

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java [hunk 1]
```diff
@@ -83,15 +83,15 @@ public void testBasicOperations() {
...
-            new MockEvent(1, 0),
-            new MockEvent(1, 1),
-            new MockEvent(1, 2),
-            new MockEvent(2, 0),
-            new MockEvent(2, 1),
-            new MockEvent(2, 3),
-            new MockEvent(3, 0),
-            new MockEvent(3, 1),
-            new MockEvent(3, 2)
+                new MockEvent(1, 0),
```
Comment (dajac) [inline coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: Why did you change this? We use 4 spaces everywhere in this module.

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java [hunk 2]
```diff
@@ -118,9 +118,9 @@ public void testAddFirst() {
         EventAccumulator<Integer, MockEvent> accumulator = new EventAccumulator<>();
 
         List<MockEvent> events = Arrays.asList(
-            new MockEvent(1, 0),
-            new MockEvent(1, 1),
-            new MockEvent(1, 2)
+                new MockEvent(1, 0),
```
Comment (dajac) [inline coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: ditto.

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java [hunk 3]
```diff
@@ -83,15 +83,15 @@ public void testBasicOperations() {
         assertNull(accumulator.poll());
 
         List<MockEvent> events = Arrays.asList(
```
Comment (dajac) [inline coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: We could use List.of here.

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java [hunk 4]
```diff
@@ -118,9 +118,9 @@ public void testAddFirst() {
         EventAccumulator<Integer, MockEvent> accumulator = new EventAccumulator<>();
 
         List<MockEvent> events = Arrays.asList(
```
Comment (dajac) [inline coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: List.of.

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java [hunk 5]
```diff
@@ -380,10 +380,9 @@ public Map<String, Integer> extractGroupConfigMap(ShareGroupConfig shareGroupCon
      * Copy the subset of properties that are relevant to consumer group.
      */
     public Map<String, Integer> extractConsumerGroupConfigMap() {
-        Map<String, Integer> groupProps = new HashMap<>();
-        groupProps.put(GroupConfig.CONSUMER_SESSION_TIMEOUT_MS_CONFIG, consumerGroupSessionTimeoutMs());
-        groupProps.put(GroupConfig.CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG, consumerGroupHeartbeatIntervalMs());
-        return Collections.unmodifiableMap(groupProps);
+        return Map.of(
+                GroupConfig.CONSUMER_SESSION_TIMEOUT_MS_CONFIG, consumerGroupSessionTimeoutMs(),
+                GroupConfig.CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG, consumerGroupHeartbeatIntervalMs());
```
Comment (dajac) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: nit: To follow the style in the module.
```
return Map.of(
    GroupConfig.CONSUMER_SESSION_TIMEOUT_MS_CONFIG, consumerGroupSessionTimeoutMs(),
    GroupConfig.CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG, consumerGroupHeartbeatIntervalMs()
);
```

**Review by dajac — COMMENTED** [review]
@mimaison Thanks for the patch. I left a few nits for consideration.

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java [hunk 6]
```diff
@@ -120,10 +120,10 @@ public class GroupCoordinatorConfig {
 
     public static final String CONSUMER_GROUP_ASSIGNORS_CONFIG = "group.consumer.assignors";
     public static final String CONSUMER_GROUP_ASSIGNORS_DOC = "The server side assignors as a list of full class names. The first one in the list is considered as the default assignor to be used in the case where the consumer does not specify an assignor.";
-    public static final List<String> CONSUMER_GROUP_ASSIGNORS_DEFAULT = Collections.unmodifiableList(Arrays.asList(
+    public static final List<String> CONSUMER_GROUP_ASSIGNORS_DEFAULT = List.of(
             UniformAssignor.class.getName(),
             RangeAssignor.class.getName()
```
Comment (dajac) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: The indentation is actually wrong here too.

**Review by m1a2st — COMMENTED** [review]
Hello @mimaison, Thanks for this PR, left a comment.
Should we change `CoordinatorLoader.LoadSummary` and `CoordinatorRecord` to record class?
https://github.com/apache/kafka/blob/d8bfbb7d1a1c262b93d40d8d582a96b21ba30e7d/coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRecord.java#L30
https://github.com/apache/kafka/blob/d8bfbb7d1a1c262b93d40d8d582a96b21ba30e7d/coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorLoader.java#L52

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java (hunk 3 above)
Comment (mimaison) [reply coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: I kept `Arrays.asList()` for consistency with the other test below that requires a mutable list.

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java (hunk 1 above)
Comment (mimaison) [reply coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: My bad, IntelliJ must have reformatted that, reverting these changes

File: coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java (hunk 4 above)
Comment (mimaison) [reply coordinator-common/src/test/java/org/apache/kafka/coordinator/common/runtime/EventAccumulatorTest.java]: We do `Collections.reverse()` below which requires a mutable list.

**Comment by mimaison** [comment]
@m1a2st We've not started adopting `record` classes anywhere in the project yet. I think we should start using that feature but we can do it in another PR.

**Review by dajac — APPROVED** [review]
LGTM, thanks.

**Review by m1a2st — APPROVED** [review]
